elan/mux/mux.go

```diff
@@ -0,0 +1,110 @@
+// Package mux provides a multiplexer over two gocloud buckets.
+package mux
+
+import (
+	"context"
+	"io"
+
+	"github.com/peterebden/go-cli-init/v2"
+	"gocloud.dev/blob"
+)
+
+var log = cli.MustGetLogger()
+
+// New returns a new mux.
+func New(primary, secondary *blob.Bucket) *Mux {
+	return &Mux{
+		primary:   primary,
+		secondary: secondary,
+	}
+}
+
+type Mux struct {
+	primary, secondary *blob.Bucket
+}
+
+func (m *Mux) Exists(ctx context.Context, key string) (bool, error) {
+	if exists, err := m.primary.Exists(ctx, key); exists && err == nil {
+		return true, nil
+	}
+	return m.secondary.Exists(ctx, key)
+}
```

Comment:
Could this theoretically say something doesn't exist because of a transient error with the primary? I think if the primary returns an error, we should return an error rather than checking the secondary.

Reply:
Yeah fair enough, I did wonder about this. Changed.